server/historian/Dockerfile

```
@@ -1,15 +1,15 @@
 # Copyright (c) Microsoft Corporation and contributors. All rights reserved.
 # Licensed under the MIT License.
 
-FROM node:12.22.7-stretch-slim AS base
+FROM node:14.18.1-slim AS base
```

```suggestion
FROM node:14.18.1-stretch-slim AS base
```

@znewton @yunho-microsoft I assume this should be stretch-slim here, even though we're moving to 14? Should gitrest be updated for consistency?

---

I changed to stretch-slim because there is a secure fix in stretch but not in slim. Here I think slim is fine. @znewton What do you think?

---

The original security issue was [CVE-2018-20482](https://nvd.nist.gov/vuln/detail/cve-2018-20482). I guess the question is whether 14.18.1-slim has tar >1.30

---

I checked 12.22.7-slim, 12.22.7-stretch-slim, 14.18.1-slim, and 14.18.1-stretch-slim TAR versions using

```dockerfile
FROM node:12.22.7-stretch-slim AS base

RUN tar --version
```
```shell
docker build . --progress "plain" --no-cache
```

All output `tar (GNU tar) 1.29` which is odd because our original Component Governance Alert stated the problem in 12.22.7-slim was fixed in 12.22.7-stretch-slim, but they all show the same version :/

---

I think leave it as 14.18.1-slim

---

@Zach The alert said 1.29b-1.1 is vulnerable but 1.29b-1.1+deb9u1 is fine (https://security-tracker.debian.org/tracker/CVE-2018-20482). Do you have the detail version number?

---

Looks like [14.18.2](https://nodejs.org/en/blog/release/v14.18.2/) fixes a CNAME regression

---

@znewton @yunho-microsoft Can you guys take a fresh look at the Dockerfile changes? I think I captured everything here including the update to 14.18.2, but please double-check.